ansible_collections/arista/avd/docs/semantic-versioning.md

```diff
@@ -0,0 +1,85 @@
+| `eos_designs`<br>Documentation | ✅ | ✘ | Fabric documentation artifacts may change between minor releases.<br>Breaking changes to CSV output will be communicated in the release notes. |
+| `eos_designs`<br>`avd_switch_facts` or `switch.*` | ✅ | ✘ | The `switch.*` and `avd_switch_facts` are internal datamodels used for sharing data between various stages of `eos_designs`. They **should not** be used in custom templates for interface descriptions or IP addressing, since they are subject to change without notice. |
+| `eos_cli_config_gen`<br>EOS configuration | ✅ | ✅ | EOS configuration generated by `eos_cli_config_gen` follows SemVer.<br>Breaking changes may occur in rare cases when addressing a bug.<br>Reordering of the CLI output may occur but without impacting the resulting configuration on EOS. |
+| `eos_cli_config_gen`<br>Documentation | ✅ | ✘ | Device documentation artifacts may change during minor releases.|
+| `cvp_configlet_upload` | ✅ | ✅ | |
+| `eos_config_deploy_cvp` | ✅ | ✘ | Structured configuration output may change during minor releases to accommodate changes in the ansible-cvp collection. <br>The resulting CloudVision configuration will be maintained, unless in rare cases, when addressing a bug.<br>Breaking changes will be communicated in the release notes. |
+| `cv_deploy` | ✅ | ✅ | The resulting CloudVision configuration will be maintained, unless in rare cases, when addressing a bug. |
+| `eos_config_deploy_eapi` | ✅ | ✅ | |
+| `eos_validate_state` | ✅ | ✘ | The generated reports and other artifacts may change between minor releases.<br>Breaking changes to CSV outputs will be communicated in the release notes.|
+| `dhcp_provisioner` | ✅ | ✘ | Structured configuration output may change during minor releases to accommodate changes in the ansible-cvp collection.<br>The resulting DHCP configuration will be maintained, unless in rare cases, when addressing a bug. |
+| `build_output_folders` | ✅ | ✅ | |
+
+### Plugins
+
+| Plugins | Inputs Follow SemVer | Outputs Follow SemVer | Notes |
+| ------- | :------------------: | :-------------------: | ----- |
+| Filter: `add_md_toc` | ✅ | ✅ | |
+| Filter: `convert_dicts` | ✅ | ✅ | |
+| Filter: `decrypt` | ✅ | ✅ | |
+| Filter: `default` | ✅ | ✅ | |
+| Filter: `encrypt` | ✅ | ✅ | |
+| Filter: `generate_esi` | ✅ | ✅ | |
+| Filter: `generate_lacp_id` | ✅ | ✅ | |
+| Filter: `generate_route_target` | ✅ | ✅ | |
+| Filter: `hide_passwords` | ✅ | ✅ | |
+| Filter: `is_in_filter` | ✅ | ✅ | |
+| Filter: `list_compress` | ✅ | ✅ | |
+| Filter: `natural_sort` | ✅ | ✅ | |
+| Filter: `range_expand` | ✅ | ✅ | |
+| Filter: `snmp_hash` | ✅ | ✅ | |
+| Filter: `status_render` | ✅ | ✅ | |
+| Action: `batch_template` | ✘| ✘ | Internal plugin, not intended for external use. |
```

Comment:
This would be used by others, so we should support this under SemVer.

Reply:
addressed in fbb906d